Content.Server/Psionics/PsionicsSystem.Events.cs

```diff
@@ -0,0 +1,19 @@
+
+namespace Content.Server.Psionics
+{
+    /// <summary>
+    ///     Raised on an entity about to roll for a Psionic Power, after their baseline chances of success are calculated.
+    /// </summary>
+    [ByRefEvent]
+    public struct OnRollPsionicsEvent
+    {
+        public readonly EntityUid Roller;
+        public float BaselineChance;
+        public OnRollPsionicsEvent(EntityUid roller, float baselineChance)
+        {
+            Roller = roller;
+            BaselineChance = baselineChance;
+        }
+    }
```

Comment:
You should be good to switch this to a record Struct- they're generally nicer. Check the syntax elsewhere and keep in mind one the fields in readonly. (Shouldn't be a problem?)